plugins/notifications-backend/src/service/handlers.ts

```diff
@@ -239,6 +239,60 @@ export async function getNotificationsCount(
   return ret;
 }
 
+export async function setRead(
+  dbClient: Knex<any, any>,
+  messageID: string,
+  user: string,
+  read: boolean,
+) {
+  let isUpdate = false;
+  let isInsert = false;
+
+  // verify that message id exists
+  await dbClient('messages')
+    .where('id', messageID)
+    .count('* as CNT')
+    .then(count => {
+      const msgcount = count[0].CNT;
+
+      if (msgcount !== '1') {
+        throw new Error(`message ID ${messageID} does not exist`);
+      }
+    });
+
+  // check user row exists
+  await dbClient('users')
+    .where('message_id', messageID)
+    .andWhere('user', user)
+    .select('*')
+    .then(rows => {
+      if (!Array.isArray(rows)) {
+        return;
+      }
+      if (rows.length === 1) {
+        isUpdate = rows[0].read !== read;
+      } else if (rows.length === 0) {
+        isInsert = true;
+      }
+    });
+
+  // insert/update user row
+  if (isInsert) {
+    await dbClient('users').insert({
+      message_id: messageID,
+      user: user,
+      read: read,
+    });
+  }
+
+  if (isUpdate) {
+    await dbClient('users')
```

Comment:
As a side-effect (probably not to be fixed now, just to be aware of it), the user can be added among recipients just by knowing the message ID. No matter the user is matching targetGroups  or targetUsers of message creation.

Reply:
of course. security wise i am not concerned. but there could be an accident here. i am waiting for Christophe to reply regarding detecting the requesting/calling user.
I hope we will get rid of the use parameter everywhere.
but even if we do get rid of the user parameter we still have the problem you describe. User A can say: mark messgae-for-user-B as read by A.
We can add validation if you think it is required. i don't mind adding it

Reply:
I would not drive API definition by that.

There is none such converter natively. I believe there will be a few implementations in npm libraries.
Anyway, we can go with something like:
```
    const toBoolean = (str?: string) => {
      if (!str) {
        return undefined;
      }
      return str === 'true' ? true : false;
    }
```